### DISCUSSION TITLE
How to change order of panes

### DISCUSSION THREAD

**Discussion by c02y** [jesseduffield/lazygit#4009] [Q&A]
By default, the order of panes is:
- [1] Status
- [2] Files
- [3] Local branches
- [4] Commits
- [5] Stash

I can use `Tab` and `Shift-Tab` to switch back and forth among all the panes or use `1/2/3/4/5` numbers to the specific pane.

But can I change the order of the panes, such as putting `Commits` pane after `Files` so `Commits` becomes [3]?

**Comment by stefanhaller** [comment]
No, that's not possible right now.

A few months ago I had the same thought, I'd prefer the order to be Files, Commits, Branches. I made a branch that implements that (simply hard-coded, not configurable), and used it for a few days to see if I could get used to it. I couldn't; the current layout was so ingrained in my muscle memory (or rather, where my eyes move to find information) that I couldn't retrain myself.

But I agree that for new users it might be better to order them this way, or to make it configurable. I'll see if I can dust off that branch to see how much effort that would be.

**Comment by lonix1** [comment]
This would be a nice configurable option for new users. Other than work-related repos, most of mine are local, so for me it makes sense to move "Local branches" to the bottom.

Also it would be nice to make it much shorter, as we have only a few branches, so it's a mostly-empty pane which takes lots of screen real estate.

**Reply by c02y** [reply]
BTW, you can start lazygit with `lazygit branch` which you will focus on the branch right away, just in case you don't know that.

And I cannot agree more that the pane height can be auto-adjusted according to the actual content (maybe set a max value for it in config file), most of the time my `Local branches` pane is one or two lines but it takes a lot of space, but I have to scroll in Files pane to see all the content, although I can use `+` to maximize the pane, shrinking the size of small/empty pane will be better.

**Comment by sadleb** [comment]
Thanks for the awesome project! Just adding a "seconded", making the pane order configurable so that Commits is before Local branches would be nice for new users (aka me).
